hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/InconsistentS3ClientFactory.java

```diff
@@ -44,8 +46,17 @@ protected AmazonS3 buildAmazonS3Client(
     LOG.warn("** FAILURE INJECTION ENABLED.  Do not run in production! **");
     LOG.warn("List inconsistency is no longer emulated; only throttling and read errors");
     InconsistentAmazonS3Client s3
```

Comment:
@passaro I think you've done some work for this, so I've just made some temp changes to get this PR to work. And will update once I'm able to rebase to your latest changes.